### DISCUSSION TITLE
logging - Reduktion der Einträge

### DISCUSSION THREAD

**Discussion by faulgurke** [e-netze/webgis-community#131] [Q&A]
Wir haben sowohl in der portal.config (<add key="trace" value="false" /> etc.) als auch in der api.config (whitepaper Logging) Einstellungen zum Logging vorgenommen.
Allerdings werden nach wie vor Unmengen von Log-Einträgen produziert.
An welcher Schraube (bzw. Schrauben) muss man drehen, damit wir das reduzieren können?

**Comment by gis-eni** [comment]
Es wäre natürlich interessant, was ihr für Einstellungen getroffen habt, und welche Erwartungen ihr beim Logging habt. 

Was wird geloggt?

Steht in der api.config oder portal.config irgendwo ``<add key="trace" value="true" />``?

**Reply by faulgurke** [reply]
Die Erwartung ist, dass innerhalb des Portals der Benutzer, die Rolle (mit Rollenparameter) und die email geloggt werden.
In der api.config sieht das so aus:
<add key="logging-type" value="files" />
<add key="logging-log-performance" value="true" />
 <add key="Log_Performance_Columns" value="X-AUTHORIZE-roles;X-AUTHENTICATE-mail,X-AUTHENTICATE-gvOuOKZ" />
    <add key="logging-log-exceptions" value="true" />
    <add key="Log_Path" value="D:\webgis5_pvp\logs" /> 

In der portal.config sieht das so aus:

 
 <add key="portal_extended_role_parameters_header" value="X-AUTHENTICATE-mail, X-AUTHORIZE-roles"/>
 <add key="portal_ticket_parameter" value="X-AUTHORIZE-roles"/>
 **<add key="trace" value="true" />**
 <add key="tracePath" value="D:\webgis5_pvp\logs" />

Im Moment habe ich in der portal.config **trace auf false** gestellt, weil die Log-Files riesig geworden sind. Dieselben Einträge sind mehrfach vorhanden. 
Die Frage ist, ob ich die Einträge in beiden Konfigurationsdateien brauche.

**Reply by gis-eni** [reply]
Trace ist auch nicht zum Logging gedacht

**Reply by faulgurke** [reply]
Wieso fragst du dann, ob in der api.config oder portal.config irgendwo <add key="trace" value="true" /> steht?

**Reply by gis-eni** [reply]
???

**Reply by faulgurke** [reply]
Sorry, da hat's was gefressen - sollte so lauten:
Wieso fragst du dann, ob in der api.config oder portal.config irgendwo add key="trace" value="true" steht?
Ich war der Meinung, dass das das Logging aktiviert.

**Reply by gis-eni** [reply]
Nein, dient eigentlich nur debugging zwecken. Da kannst du quasi mitlesen, welche Headervariaben (PVP) daherkommen usw.
Für das Logging gibt es andere Einträge in der api.config. Ich nur leider auch nicht, was du da eingestelt hast. Kannst mir das auch per Mail schicken, wenn du das nicht hier rein kopieren willst

**Reply by faulgurke** [reply]
Auf das Angebot komme ich gerne zurück. Vielen Dank!

**Comment by gis-eni** [comment]
Hallo Jürgen,

du musst log-performance auf „true“ setzen:

![image](https://github.com/gis-eni/webgis-forum/assets/62870568/ab89944f-e40a-4e63-920b-a773296613e9)

Dann wird auch ein webgis_performance.csv erstellt:

![image](https://github.com/gis-eni/webgis-forum/assets/62870568/f62d9481-b112-4c46-bbde-501543e7455b)

Achtung: es wird nicht dauernd geschrieben, sondern nur so alle 5min. Das File taucht also nicht gleich auf, wenn du das aktivierst. Allerdings kannst du api/instance/logging aufrufen und auf ``reload all settings...`` klicken:

<img width="274" alt="image" src="https://github.com/gis-eni/webgis-forum/assets/62870568/31fa9cac-4260-4a52-a2a0-8d080024dffc">

Dann wird das File erstellt.
